more-info/under-the-hood.mdx

```diff
@@ -4,31 +4,41 @@ description: "If you are interested in how we make the tests work, this is your
 icon: "engine"
 ---
 
-## Test case building
+## AI agents discovering and generating test steps
 
 A web app is typically composed of user flows. A user flow lets a user accomplish a certain goal. "Sign-in with email & password" is an example.
 To perform a sign-in with email & password, a certain sequence of steps - `interactions` - is required.
-We are recording and storing the interaction chain of a test case in an intermediate representation.
-The corresponding [Playwright](https://playwright.dev/) code which is exercising the UI is generated on the fly just before test execution. The interaction chain can be examined in the test detail view and the Playwright trace viewer.
 
-## Auto-maintenance
+Our AI agents mimic human users (i.e., clicks input fields, signs up for newsletter) to navigate apps, interpret app intent, and identify all relevant user flows. We are recording and storing the interaction chain of a test case in an intermediate representation.
+
+## Playwright generates test code
+
+After we record and store each test case's interaction chain, we generate the corresponding [Playwright code](https://playwright.dev/) deterministically on the fly immediately prior to test execution.
+
+Playwright code is exercising the UI. The interaction chain can be examined in the test detail view and the Playwright trace viewer.
```

Comment:
"Playwright code is exercising the UI" ?!? dont understand

Reply:
That's a roedler formulation.  suggest a different formulation,.

Reply:
Sorry no idea, maybe we should himself ...